python/tskit/trees.py

```diff
@@ -3448,6 +3449,8 @@ def trait_covariance(self, W, windows=None, mode="site", span_normalise=True):
             window (defaults to True).
         :return: A ndarray with shape equal to (num windows, num statistics).
         """
+        if W.shape[0] != self.num_samples:
+            raise ValueError("First trait dimension must be equal to number of samples.")
```

Comment:
Is this not caught by the low-level code, or are you giving a more user-friendly error message? (More user friendly message is good!)

Reply:
The second thing: the low level error is quite cryptic:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/peter/projects/tskit-dev/tskit/python/tskit/trees.py", line 3584, in trait_regression
    mode=mode, span_normalise=span_normalise)
ValueError: object of too small depth for desired array
```
... although we should actually fix this for all the stats at once.

Reply:
Hm, actually, this is just in correlation and covariance also; checking like this is fine.